virtualizarr/manifests/store.py

```diff
@@ -137,34 +133,92 @@ def parse_manifest_index(key: str, chunk_key_encoding: str = ".") -> tuple[int,
     return tuple(int(ind) for ind in parts[1].split(chunk_key_encoding))
 
 
-def find_matching_store(stores: StoreDict, request_key: str) -> StoreRequest:
-    """
-    Find the matching store based on the store keys and the beginning of the URI strings,
-    to fetch data from the appropriately configured ObjectStore.
+def _find_bucket_region(bucket_name: str) -> str:
+    import requests
 
-    Parameters:
-    -----------
-    stores : StoreDict
-        A dictionary with URI prefixes for different stores as keys
-    request_key : str
-        A string to match against the dictionary keys
+    resp = requests.head(f"https://{bucket_name}.s3.amazonaws.com")
```

Comment:
Not sure if you're calling this from a sync or async context, but if it's an async context you should make this an async request.